pkg/build/dockerfile-generator.go

```diff
@@ -143,11 +151,24 @@ func (g *DockerfileGenerator) getBaseDockerfile() ([]string, error) {
 }
 
 func (g *DockerfileGenerator) buildPorterSection() []string {
-	return []string{
-		// Remove the user-provided Porter manifest as the canonical version
-		// will migrate via its location in .cnab
-		`RUN rm $BUNDLE_DIR/porter.yaml`,
+	// The user-provided manifest may be located separate from the build context directory.
+	// Therefore, only add lines to remove it from BUNDLE_DIR if its filepath
+	// contains the current working directory.
+	if strings.Contains(g.ManifestPath, g.Getwd()) {
+		manifestSubpaths := strings.SplitAfterN(g.ManifestPath, g.Getwd(), 2)
+		if len(manifestSubpaths) != 2 {
+			return []string{}
+		}
+		manifestSubpath := manifestSubpaths[1]
+		if exists, _ := g.FileSystem.Exists(manifestSubpath); exists {
```

Comment:
Is manifest subpath supposed to be the relative path to the manifest in the current directory? I think if we use filepath.Rel for both the if statement and the logic here we can improve the readability of this code. What do you think?

Reply:
Yup, love the suggestion to use filepath.Rel -- way cleaner.